sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -357,8 +357,15 @@ ESIMD_INLINE
     __esimd_scatter_scaled<PromoT, N, decltype(si), TypeSizeLog2, scale>(
         mask.data(), si, glob_offset, offsets.data(), promo_vals.data());
   } else {
-    __esimd_scatter_scaled<T, N, decltype(si), TypeSizeLog2, scale>(
-        mask.data(), si, glob_offset, offsets.data(), vals.data());
+    using Treal = __raw_t<T>;
+    if constexpr (!std::is_same_v<Treal, T>) {
+      simd<Treal, N> Values = vals.template bit_cast_view<Treal>();
```

Comment:
I suggested to use sycl::bit_cast in my [comment](https://github.com/intel/llvm/pull/6828#discussion_r977272809), but I see bit_cast_view is used instead. Could you please explain the reason?

`bit_cast_view` is more stressful to compiler, as it involves a chain of calls/constructors and conversions, which can increase compile time and yet there is no guarantee compiler will optimize away everything. `sycl::bit_cast` should be more efficient here in the library code, esp. when it maps directly to `std::bit_cast` with C++20.

Reply:
As we checked IR, there is no difference between bit_cast_view and bit_cast, I think we can live with bit_cast_view then, since this function's code is stable and we unlikely will get bad optimization.

Approving.